libethereum/BlockChain.cpp

```diff
@@ -1430,7 +1430,11 @@ Block BlockChain::genesisBlock(OverlayDB const& _db) const
 {
 	h256 r = BlockHeader(m_params.genesisBlock()).stateRoot();
 	if (_db.exists(r))
-		return Block(*this, _db, r);
+	{
+		Block gen(*this, _db, r);
+		gen.m_currentBlock = BlockHeader(m_params.genesisBlock());
```

Comment:
This expression is executed a couple lines above. Please do not evaluate it again.